CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+# Contributing
+
+**Table of Contents**
+1. [Welcome](#welcome) (2nd draft)
+2. [Philosophy](#philosophy) (TBD)
+3. [Reporting an Issue](#reporting-an-issue) (TBD)
+4. [Fixing an Issue (PRs)](#fixing-an-issue-prs) (2nd draft)
+5. [Setup and Run](#setup-and-run) (TBD)
+
+
+## Welcome
+Hi, and welcome to Highlight.js. We're a code-highlighting engine etc., but you probably know that. You want to contribute, and this document will be your guide.
+
+There are many ways to contribute to Highlight.js. Reporting issues is super useful. Feel free to work on an open issue. You can also help by writing [a plugin](https://highlightjs.readthedocs.io/en/latest/plugin-api.html), adding [a 3rd party grammar](https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md), or creating [a theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html).
+
+In terms of getting help, GitHub issues are the primary method of communication. We're exploring the idea of creating a Gitter chat room.
+
+
+## Philosophy
+
+Highlight.js makes a strong effort to keep the core engine, it's usage, and it's feature set clean and simple.  We aim to support the happy path (the 90%), not the often innumerable edge cases (the last 10%).  When you're on the happy path everything "just works" with a single line of code.  We do our best to allow those familiar with JavaScript to easily handle the remaining 10% via small extensions or plug-ins.
+
+Feature requests are always welcome but many times the answer is:
+
+> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.
+
+If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
```

Comment:
This gives the reader a good overview of features. Maybe give it a heading, so we can link to it elsewhere, eg. `#prerequisites`. This would mean the next section needs a heading too, so it's not under `#feature-requests` (faq?)
```suggestion
### Feature Requests
Feature requests are always welcome, but many times the answer is:

> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.

If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
```

Reply:
Actually, `#reporting-an-issue` looks empty. Maye move the feature stuff there.

Reply:
I think of feature requests and issues as entirely different things.

Reply:
I think feature requests makes a great section.  I mean we have a table of contents so as long as this doesn't get too long I'm ok with it having some meat.